R/dv8.R

```diff
@@ -0,0 +1,215 @@
+#' Get git log from local repository
+#'
+#' @param git_repo_path path to local repository
+#' @param git_numstat_path file name of git log output
+#'
+#' @return path to git log output
+#' @export
+#'
+dv8_gitlog_to_gitnumstat <- function(git_repo_path, git_numstat_path) {
+  setwd(git_repo_path)
+  system2(command = "git",
+          args = c("log", "--numstat", "--date=iso"),
+          stdout = git_numstat_path
+          )
+  return(git_numstat_path)
+}
+
+
+#' Converts a git log text file into a dependency matrix file
+#'
+#' An input file *.txt is converted to *.dv8-dsm
+#'
+#' @param git_numstat_path path to local git log text file
+#' @param start start of date/time range in ISO format ie. 2017-07-08T00:00:00Z
+#' @param stop end of date/time range in ISO format ie. 2018-07-08T00:00:00Z
+#' @param max_cochange_count maximum count of co-changed files per commit, default 1000
+#' @param hdsmb_path name of output dependency matrix file, defaults to name of input
+#' @param params_output_file output file used to record parameters
+#'
+#' @return path to dependency matrix file
+#' @export
+#'
+dv8_gitnumstat_to_hdsmb <- function(git_numstat_path,
+                                    hdsmb_path,
+                                    start,
+                                    stop,
+                                    max_cochange_count=1000,
+                                    params_output_file="") {
+
+  output_file_arg <- sprintf("-outputFile %s", hdsmb_path)
+  start_arg <- sprintf("-start %s", start)
+  stop_arg <- sprintf("-stop %s", stop)
+  max_cochange_count_arg <- sprintf("-maxCochangeCount %i", max_cochange_count)
+  params_output_file_arg <- ""
+
+  if (params_output_file != "") {
+    params_output_file_arg <- sprintf("-paramsOutputFile %s ", params_output_file)
+  }
+
+  cmd <- stri_c("dv8-console", " ",
+                "scm:history:gittxt:convert-matrix", " ",
+                start_arg, " ",
+                stop_arg, " ",
+                output_file_arg, " ",
+                max_cochange_count_arg, " ",
+                params_output_file_arg,
+                git_numstat_path)
+
+  termId <- rstudioapi::terminalExecute(cmd, env=c('PATH="/home/vboxuser/DV8/bin:$PATH"'))
+  while (is.null(rstudioapi::terminalExitCode(termId))) {
+    Sys.sleep(0.1)
+  }
+
+  return(hdsmb_path)
+}
+
+
+#' Calculates the Decoupling Level from dependency matrix
+#'
+#' @param hsdsmb_path path to dependency matrix file
+#' @param json_path name of output file
+#'
+#' @return path of output file
+#' @export
+#'
+dv8_hsdsmb_to_decoupling_level <- function(hsdsmb_path, json_path) {
+  output_file_arg <- sprintf("-outputFile %s ", json_path)
+
+  cmd <- stri_c("dv8-console", " ",
```

Comment:
So just to be clear, every function that uses dv8-console command in rstudioapi should have a dv8_path parameter? So for this specific example, it would look something like

```r
dv8_hsdsmb_to_decoupling_level <- function(dv8_path, hsdsmb_path, json_path) {
  output_file_arg <- sprintf("-outputFile %s ", json_path)
  dv8_path <- path.expand(dv8_path)
  cmd <- stri_c(dv8_path, " ",
...